libraries/AP_HAL/I2CDevice.h

```diff
@@ -77,7 +77,10 @@ class I2CDevice : public Device {
 class I2CDeviceManager {
 public:
     /* Get a device handle */
-    virtual OwnPtr<AP_HAL::I2CDevice> get_device(uint8_t bus, uint8_t address) = 0;
+    virtual OwnPtr<AP_HAL::I2CDevice> get_device(uint8_t bus, uint8_t address,
+                                                 uint32_t bus_clock=400000,
+                                                 bool use_smbus = false,
```

Comment:
Maybe this should be an enum of modes `I2C` `SMBUS` etc rather then a bool? It would just make the calling locations a lot cleaner looking.